sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+    });
+    const ids = studentRestrictionIds.map(
+      (studentRestrictionId) => studentRestrictionId.id,
+    );
+    await db.studentRestriction.delete({ id: In(ids) });
+  });
+
+  it("Should save only one legacy restriction for the student if one or more restrictions are imported from SFAS which do not have a mapping to the SIMS restrictions", async () => {
+    const studentExists = await db.student.findOne({
+      where: {
+        birthDate: getISODateOnlyString(new Date("1998-03-24")),
+        user: { lastName: "FOUR" },
+        sinValidation: { sin: "900041310" },
+      },
+    });
+    let student = studentExists;
+    if (!studentExists) {
+      student = await saveFakeStudent(db.dataSource);
+      // Update the student to ensure that the student imported from SFAS is the same student as the one created above.
+      student.birthDate = getISODateOnlyString(new Date("1998-03-24"));
+      student.user.lastName = "FOUR";
+      student.sinValidation.sin = "900041310";
+      await db.student.save(student);
+    }
+    // Queued job.
+    const job = createMock<Job<void>>();
+    mockDownloadFiles(sftpClientMock, [SFAS_LEGACY_RESTRICTION_FILENAME]);
+    // Act
+    await processor.processSFASIntegrationFiles(job);
+    // Assert
+    // Expect the file was deleted from SFTP.
+    expect(sftpClientMock.delete).toHaveBeenCalled();
+    // Expect only one Legacy restriction to be inserted.
+    const studentRestrictionsCount = await db.studentRestriction.count({
+      where: {
+        student: { id: student.id },
+        restriction: { restrictionCode: "LGCY" },
+      },
+    });
+    expect(studentRestrictionsCount).toBe(1);
+  });
+
+  it(
+    "Should not insert a restriction to SIMS student restrictions if the same restriction is imported from SFAS but is already active in SIMS, " +
+      "Should not insert a legacy restriction if a legacy restriction is already active in SIMS for the student, " +
+      "Should insert a restriction after mapping it to SIMS restriction if it is either inactive or is not at all present in the SIMS student restrictions, " +
+      "Should not insert a restriction if it has a restriction resolution date.",
+    async () => {
+      const studentExists = await db.student.findOne({
+        where: {
+          birthDate: getISODateOnlyString(new Date("1998-03-24")),
+          user: { lastName: "FOUR" },
+          sinValidation: { sin: "900041310" },
+        },
+      });
+      let student = studentExists;
+      if (!studentExists) {
+        student = await saveFakeStudent(db.dataSource);
+        // Update the student to ensure that the student imported from SFAS is the same student as the one created above.
+        student.birthDate = getISODateOnlyString(new Date("1998-03-24"));
+        student.user.lastName = "FOUR";
+        student.sinValidation.sin = "900041310";
+        await db.student.save(student);
+      }
+      const restrictionB6B = await db.restriction.findOne({
```

Comment:
Can create a small private method for the test `saveStudentRestriction(restrictionCode, studentId, isActive)`

Reply:
Didn't get this one.

Reply:
Created two private methods for `find` and `save` operations.

Reply:
Why it cannot be simplified as one private method??

Reply:
Created one method to `find` and `save`.